internal/metamorph/processor.go

```diff
@@ -305,34 +289,73 @@ func (p *Processor) StartProcessStatusUpdatesInStorage() {
 			case <-p.ctx.Done():
 				return
 			case statusUpdate := <-p.storageStatusUpdateCh:
-				// Ensure no duplicate hashes, overwrite value if the status has higher value than existing status
-				foundStatusUpdate, found := statusUpdatesMap[statusUpdate.Hash]
-				if !found || (found && statusUpdate.Status > foundStatusUpdate.Status) {
-					statusUpdatesMap[statusUpdate.Hash] = statusUpdate
-				}
+				// Ensure no duplicate statuses
+				updateStatusMap(statusUpdatesMap, statusUpdate)
 
 				if len(statusUpdatesMap) >= p.processStatusUpdatesBatchSize {
-					p.CheckAndUpdate(statusUpdatesMap)
+					p.checkAndUpdate(statusUpdatesMap)
 					statusUpdatesMap = map[chainhash.Hash]store.UpdateStatus{}
 				}
 			case <-ticker.C:
 				if len(statusUpdatesMap) > 0 {
-					p.CheckAndUpdate(statusUpdatesMap)
+					p.checkAndUpdate(statusUpdatesMap)
 					statusUpdatesMap = map[chainhash.Hash]store.UpdateStatus{}
 				}
 			}
 		}
 	}()
 }
 
-func (p *Processor) statusUpdateWithCallback(statusUpdates []store.UpdateStatus) error {
-	updatedData, err := p.store.UpdateStatusBulk(context.Background(), statusUpdates)
+func (p *Processor) checkAndUpdate(statusUpdatesMap map[chainhash.Hash]store.UpdateStatus) {
+	if len(statusUpdatesMap) == 0 {
+		return
+	}
+
+	statusUpdates := make([]store.UpdateStatus, 0, p.processStatusUpdatesBatchSize)
+	doubleSpendUpdates := make([]store.UpdateStatus, 0)
+
+	for _, status := range statusUpdatesMap {
+		if len(status.CompetingTxs) > 0 {
```

Comment:
Wouldn't this be a bit more explicit?

```
status.Status == metamorph_api.Status_DOUBLE_SPEND_ATTEMPTED
```

Reply:
That's correct, because for now only `DOUBLE_SPEND_ATTEMPTED` transactions have CompetingTxs. 

But, I can imagine, in the future, transactions having a different status (e.g. `REJECTED`) with CompetingTxs. Then, they should also be updated in the same way that double spend transactions are updated. That's why I decided on that check instead of `Status == metamorph_api.Status_DOUBLE_SPEND_ATTEMPTED`

Let me know if my reasoning makes sense. If you this use case I described will never occur, I'll change it to your suggestion.

Reply:
Yes, I think you're right. Because the only reason you want to keep updating them is not the status but the competing transactions 👍